lib/topology/platforms/shell.py

```diff
@@ -285,14 +296,15 @@ def __init__(
         self._timeout = timeout or -1
         self._encoding = encoding
         self._try_filter_echo = try_filter_echo
+        self._auto_connect = auto_connect
```

Comment:
By the way, what is the purpose of this? This can only introduce allow the user to realize that the shell is not connected when an exception is raised. How this exception can be handled in a way different than just attempting to connect again?